chat-plugins/scavengers.js

```diff
@@ -26,6 +28,42 @@ if (scavengersRoom) {
 	}
 }
 
+let scavengersLB;
+let dataFile = './config/chat-plugins/scavdata.json';
+fs.readFile(dataFile, 'utf8', function (err, contents) {
+	if (err) {
+		scavengersLB = {};
+		return;
+	}
+	scavengersLB = JSON.parse(contents);
+});
+
+function addPoints(username, points) {
+	let id = toId(username);
+	if (id in scavengersLB) {
+		scavengersLB[id].points += points;
+	} else {
+		scavengersLB[id] = {
+			points: points,
+			name: Chat.escapeHTML(username),
+		};
+	}
+}
+let scavWriteData = function () {
+	fs.writeFile(dataFile, JSON.stringify(scavengersLB));
+};
+
+let getLadder = function () {
+	let items = [];
+	for (let id in scavengersLB) {
```

Comment:
The leaderboard never really gets very large, since it resets every month (the current month leaderboard has 40-50 people), so I'm not sure that's necessary